packages/@magic-ext/farcaster/src/index.ts

```diff
@@ -1,142 +1,65 @@
-import { Extension, FarcasterLoginEventEmit, JsonRpcRequestPayload } from '@magic-sdk/commons';
-import type {
-  CreateChannelAPIResponse,
-  AuthenticateAPIResponse,
-  WatchStatusResponse,
-  AuthClientError,
-} from '@farcaster/auth-client';
+import { Extension } from '@magic-sdk/commons';
+import type { CreateChannelAPIResponse, AuthenticateAPIResponse, AuthClientError } from '@farcaster/auth-client';
 import { createAppClient, viemConnector } from '@farcaster/auth-client';
 import { FarcasterPayloadMethod } from './types';
 import { isMobile } from './utils';
 
-type Handle = {
-  on: Handler;
+const DEFAULT_SHOW_UI = true;
+
+type LoginParams = {
+  showUI: boolean;
 };
 
-const EVENT = {
-  CHANNEL: 'channel',
-  DONE: 'done',
-  ERROR: 'error',
+const FarcasterLoginEventOnReceived = {
+  OpenChannel: 'channel',
+  Success: 'success',
+  Failed: 'failed',
 } as const;
 
-type ChannelCallback = (params: CreateChannelAPIResponse) => void;
-type DoneCallback = (params: AuthenticateAPIResponse) => void;
-type ErrorCallback = (params: AuthClientError) => void;
-
-const isChannelCallback = (callback: Function): callback is ChannelCallback => typeof callback === 'function';
-const isDoneCallback = (callback: Function): callback is DoneCallback => typeof callback === 'function';
-
-interface Handler {
-  (event: 'channel', callback: ChannelCallback): Handle;
-  (event: 'done', callback: DoneCallback): Handle;
-  (event: 'error', callback: ErrorCallback): Handle;
-}
-
-const DEFAULT_RELAY_URL = 'https://relay.farcaster.xyz';
-const DEFAULT_SIWE_URI = 'https://example.com/login';
-const DEFAULT_TIMEOUT = 60000;
-const DEFAULT_INTERVAL = 500;
+type FarcasterLoginEventHandlers = {
+  [FarcasterLoginEventOnReceived.OpenChannel]: (channel: CreateChannelAPIResponse) => void;
+  [FarcasterLoginEventOnReceived.Success]: (data: AuthenticateAPIResponse) => void;
+  [FarcasterLoginEventOnReceived.Failed]: (error: AuthClientError) => void;
+};
 
 export class FarcasterExtension extends Extension.Internal<'farcaster'> {
   name = 'farcaster' as const;
   config = {};
+  appClient = createAppClient({ ethereum: viemConnector() });
+  channel: CreateChannelAPIResponse | null = null;
 
-  public login = ({ showUI }: { showUI: boolean }): Handle => {
-    const appClient = createAppClient({
-      relay: DEFAULT_RELAY_URL,
-      ethereum: viemConnector(),
-    });
-
-    const channelPromise = appClient.createChannel({
-      siweUri: DEFAULT_SIWE_URI,
-      domain: window.location.host,
-    });
-
-    const statusPromise: WatchStatusResponse = channelPromise.then(({ data }) => {
-      return appClient
-        .watchStatus({
-          channelToken: data.channelToken,
-          timeout: DEFAULT_TIMEOUT,
-          interval: DEFAULT_INTERVAL,
+  constructor() {
+    super();
+    if (isMobile()) {
```

Comment:
@ysm-dev so we only want to call this for mobile?

Reply:
changed: [9e46650](https://github.com/magiclabs/magic-js/pull/754/commits/9e4665016c300c75bac6e9d8bc7f688bfe17ba28)